manipulator_gym/utils/gym_wrappers.py

```diff
@@ -5,6 +5,59 @@
 import logging
 
 
+class CheckAndRebootJoints(gym.Wrapper):
+    """
+    Every step, check whether joints have failed and reboot them if necessary.
+    """
+    def __init__(self, env, interface, check_every_n_steps: int = 1):
+        super().__init__(env)
+        self.interface = interface
+        self.widowx_joints = [
+            "waist",
+            "shoulder",
+            "elbow",
+            "forearm_roll",
+            "wrist_angle",
+            "wrist_rotate",
+            "gripper",
+        ]
+        self.step_number = 0
+        self.every_n_steps = check_every_n_steps
+        self.motor_failed = np.zeros_like(self.action_space.sample())
+    
+    def step(self, action):
+        self.step_number += 1
+        res = self.interface.custom_fn("motor_status")
+        
+        if res is not None and self.step_number % self.every_n_steps == 0:
+            for i, status in enumerate(res):
+                if status != 0:
+                    self.motor_failed[i] = status
+                    break
+        
+        obs, reward, done, trunc, info = self.env.step(action)
+        if any(self.motor_failed):
+            trunc = True
+        
+        return obs, reward, done, trunc, info
+    
+    def reset(self, **kwargs):
+        self.step_number = 0
+        
+        # reset joints on reset
+        res = self.interface.custom_fn("motor_status")
+        if res is not None:
+            for i, status in enumerate(res):
+                if status != 0:
+                    self.interface.custom_fn("reboot_motor", joint_name=self.widowx_joints[i])
```

Comment:
should we move the robot to a sleep position (or other safer position) before rebooting? is okay if it isnt too.